src/L6/datasorter.py

```diff
@@ -33,7 +35,23 @@ def validate_input_list(input_list: list):
 
     # Exercise 25
     def set_input_data(self, file_path_name: str):
-        pass
+        if type(file_path_name) is not str:
+            raise TypeError
+        if not os.path.exists(file_path_name) and os.path.isfile(file_path_name):
+            raise FileNotFoundError
+        elif not file_path_name.lower().endswith(".csv"):
+            raise NotCSVException
+        else:
+            self.data = []
+            with open(file_path_name, newline= '') as csvfile:
+                dialect = csv.Sniffer().sniff(csvfile.read(1024), [',','|'])
+                csvfile.seek(0)
+                items = csv.reader(csvfile,dialect)
+                for row in items:
+                    for i in row:
+                        self.data.append(i)
```

Comment:
I forgot mentioning this yesterday, but, what's the type of `i` here? Is it `str`?

If it is, we need to cast that into a number, remember that the list will support both `int` and `float`.